src/physics/navier_stokes.cpp

```diff
@@ -567,38 +571,72 @@ ::dissipative_flux_templated (
 
 template <int dim, int nstate, typename real>
 void NavierStokes<dim,nstate,real>
-::boundary_face_values (
-   const int boundary_type,
-   const dealii::Point<dim, real> &pos,
+::boundary_wall (
    const dealii::Tensor<1,dim,real> &/*normal_int*/,
-   const std::array<real,nstate> &/*soln_int*/,
-   const std::array<dealii::Tensor<1,dim,real>,nstate> &/*soln_grad_int*/,
+   const std::array<real,nstate> &soln_int,
+   const std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_int,
    std::array<real,nstate> &soln_bc,
    std::array<dealii::Tensor<1,dim,real>,nstate> &soln_grad_bc) const
 {
-    if (boundary_type == 1000) {
-        // Manufactured solution boundary condition 
-        // -- TO DO: could make this a function later on, similar to how its done in euler
-        // Note: This is consistent with Navah & Nadarajah (2018)
-        std::array<real,nstate> boundary_values;
-        std::array<dealii::Tensor<1,dim,real>,nstate> boundary_gradients;
-        for (int i=0; i<nstate; i++) {
-            boundary_values[i] = this->manufactured_solution_function->value (pos, i);
-            boundary_gradients[i] = this->manufactured_solution_function->gradient (pos, i);
-        }
-        for (int istate=0; istate<nstate; istate++) {
-            soln_bc[istate] = boundary_values[istate];
-            // soln_grad_bc[istate] = soln_grad_int[istate]; // done in convection_diffusion.cpp
-            soln_grad_bc[istate] = boundary_gradients[istate];
-        }
+    using wall_temperature_boundary_condition_enum = Parameters::NavierStokesParam::WallTemperatureBoundaryConditionEnum;
+
+    // No-slip wall boundary conditions
+    // Given by equations 460-461 of the following paper
+    // Hartmann, Ralf. "Numerical analysis of higher order discontinuous Galerkin finite element methods." (2008): 1-107.
+    const std::array<real,nstate> primitive_interior_values = this->template convert_conservative_to_primitive<real>(soln_int);
+
+    // Copy density
+    std::array<real,nstate> primitive_boundary_values;
+    primitive_boundary_values[0] = primitive_interior_values[0];
+
+    if(wall_temperature_boundary_condition_type == wall_temperature_boundary_condition_enum::isothermal) { 
+        // isothermal boundary
+        primitive_boundary_values[nstate-1] = this->compute_pressure_from_density_temperature(primitive_boundary_values[0], isothermal_wall_temperature);
```

Comment:
Make sure that the isothermal_wall_temperature is being correctly non-dimensionalized.

![image](https://user-images.githubusercontent.com/9758507/163441027-7d40b831-6981-4c60-8325-26bbd6162a57.png)

Reply:
addressed in most recent commit